addons/l10n_pt/models/account_move_send.py

```diff
@@ -0,0 +1,12 @@
+from odoo import models
+
+
+class AccountMoveSend(models.TransientModel):
+    _inherit = 'account.move.send'
+
+    def _hook_invoice_document_before_pdf_report_render(self, invoice, invoice_data):
+        super()._hook_invoice_document_before_pdf_report_render(invoice, invoice_data)
+        if invoice.country_code == 'PT':
```

Comment:
We only generate the hashes when printing the document?
If it's the case what do you plan to do when generating the SAF-T (PT) report?  The hash is needed when generating the report as it is one of the report fields.